azure/iac/diagnosticsettings.rego

```diff
@@ -8,15 +8,34 @@ package rule
 
 default log_keyvault = null
 
+azure_attribute_absence ["log_keyvault"] {
+    count([c | lower(input.resources[_].type) == "microsoft.keyvault/vaults"; c := 1]) != count([c | lower(input.resources[_].type) == "microsoft.insights/diagnosticsettings"; c := 1])
+}
+
+azure_attribute_absence["log_keyvault"] {
+    resource := input.resources[_]
+    lower(resource.type) == "microsoft.insights/diagnosticsettings"
+    not contains(lower(resource.scope), "microsoft.keyvault/vaults")
+}
+
+source_path[{"log_keyvault":metadata}] {
+    resource := input.resources[i]
+    lower(resource.type) == "microsoft.insights/diagnosticsettings"
+    not contains(lower(resource.scope), "microsoft.keyvault/vaults")
+    metadata:= {
+        "resource_path": [["resources",i,"scope"]]
+    }
+}
+
 azure_attribute_absence["log_keyvault"] {
     resource := input.resources[_]
-    lower(resource.type) == "microsoft.keyvault/vaults/providers/diagnosticsettings"
+    lower(resource.type) == "microsoft.insights/diagnosticsettings"
```

Comment:
@shahinshirvani can you please check when we add `diagnosticsettings `as a child resource to `microsoft.keyvault/vaults`, how we are writing the resource type of `diagnosticsettings `? 
do we need to write as `microsoft.insights/diagnosticsettings` or `diagnosticsettings` or `microsoft.keyvault/vaults/providers/diagnosticsettings`

I cant remember but did saw `microsoft.keyvault/vaults/providers/diagnosticsettings` somewhere.

Reply:
I will check later in a separate issue